Framework/PythonInterface/test/python/plugins/algorithms/FindMultipleUMatricesTest.py

```diff
@@ -0,0 +1,148 @@
+import unittest
+
+from mantid.simpleapi import (
+    FindMultipleUMatrices,
+    AnalysisDataService,
+    IndexPeaks,
+    SetUB,
+    CreatePeaksWorkspace,
+    LoadEmptyInstrument,
+    FilterPeaks,
+    CloneWorkspace,
+    DeleteTableRows,
+)
+from mantid.kernel import V3D
+import numpy as np
+from scipy.spatial.transform import Rotation as rot
+
+
+def add_peaksHKL(peaks, Hs, Ks, Ls, iub):
+    for h in Hs:
+        for k in Ks:
+            for l in Ls:
+                pk = peaks.createPeakHKL([h, k, l])
+                if pk.getDetectorID() > 0:
+                    pk.setRunNumber(iub)  # so can keep track of UB
+                    peaks.addPeak(pk)
+
+
+class FindMultipleUMatricesTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # load empty instrument
+        cls.ws = LoadEmptyInstrument(InstrumentName="SXD", OutputWorkspace="empty_SXD")
+        axis = cls.ws.getAxis(0)
+        axis.setUnit("TOF")
+        # create a peak tables of orthorhombic domains with lattice parameters a=4, b=5, c=10
+        cls.alatt = {"a": 4, "b": 5, "c": 10, "alpha": 90, "beta": 90, "gamma": 90}
+        cls.ubs = [np.diag([1 / cls.alatt["a"], 1 / cls.alatt["b"], 1 / cls.alatt["c"]])]  # produces 7 peaks
+        cls.ubs.append(rot.from_rotvec([0, 0, 90], degrees=True).as_matrix() @ cls.ubs[0])  # produces 5 peaks
+        cls.peaks = CreatePeaksWorkspace(InstrumentWorkspace=cls.ws, NumberOfPeaks=0, OutputWorkspace=f"peaks")
+        for iub, ub in enumerate(cls.ubs):
+            SetUB(cls.peaks, UB=ub)
+            add_peaksHKL(cls.peaks, range(1, 3), range(1, 3), range(1, 4), iub)
+        # dict of default kwargs for FindMultipleUMatrices
+        cls.default_kwargs = {
+            **cls.alatt,
+            "MinDSpacing": 1.75,
+            "MaxDSpacing": 2.5,
+            "HKLTolerance": 0.15,
+            "DSpacingTolerance": 0.1,
+            "AngleTolerance": 1,
+            "Spacegroup": "P m m m",
+            "OptimiseFoundUBs": True,
+            "NumberOfUBs": 2,
+        }
+
+    @classmethod
+    def tearDownClass(self):
```

Comment:
This might be better as `cls` rather than `self`?
```suggestion
    def tearDownClass(cls):
```